src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -139,27 +156,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       response.sendRedirect("/conversations");
       return;
     }
-
+    
     String messageContent = request.getParameter("message");
 
+    UUID messageUUID = UUID.randomUUID();
+    
     // this removes any HTML from the message content
     String cleanedMessageContent = Jsoup.clean(messageContent, Whitelist.none());
     
     String cleanedAndEmojiMessage = EmojiParser.parseToUnicode(cleanedMessageContent);
 
+    Pattern hashtagPattern = Pattern.compile("(?:^|\\s|\\n)#([a-z\\d-]+)");
+    
+    Matcher matcher = hashtagPattern.matcher(cleanedAndEmojiMessage);
+    
+    Set<String> hashtags = new HashSet<String>();
```

Comment:
Group above lines together, avoid extra spacing when not necessary for clarification